src/exec/use_pty/backchannel.rs

```diff
@@ -126,15 +130,24 @@ impl ParentBackchannel {
         prefix_buf.copy_from_slice(&prefix.to_ne_bytes());
         data_buf.copy_from_slice(&data.to_ne_bytes());
 
-        self.socket.write_all(&buf)
+        if let Err(err) = self.socket.write_all(&buf) {
```

Comment:
maybe these would be more succinct written as `map_err`:

`self.socket.write_all(&buf).map_err(|e| debug_assert!(/*..*/); err)`